esp/platform/espsecurecontext.hpp

```diff
@@ -23,6 +23,9 @@
 
 class CTxSummary;
 
+static const char* const ESC_SESSION_BASED = "SessionBased";
+static const char* const ESC_AUTHENTICATED_USER_NAME = "AuthUserName";
```

Comment:
Given the potential for a manager to persist data other than a name (any user data that cannot be obtained without the password), I see two choices for this.
1. Change user name to something more generic, like user data. This makes the manager responsible for packing and unpacking whatever it needs to persist.
2. Remove the explicit reference to user name and have the session pack and unpack the map allowing managers to store and restore arbitrary properties.

Reply:
ESC_AUTHENTICATED_USER_DATA: 'username:john|password:mypwd|...'?

Reply:
If the ESP is packing and unpacking, so the manager only deals with individual values, how it's packed doesn't matter so much to the manager. If you reserve individual characters as delimiters, be sure to escape occurrences of those characters in the map.

If the manager packs the data, you shouldn't be concerned with the format.

Reply:
How about managers packs and unpacks? ESP does not have to know what is inside and just stores/sends back as a binary string.

Reply:
I can see pros and cons to both, without a clear winner. As long as I can get arbitrary data in and out of the session, I don't care.